cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -265,6 +265,7 @@ public KafkaAssemblyOperator(Vertx vertx, PlatformFeaturesAvailability pfa,
         reconcileState.initialStatus()
                 .compose(state -> state.reconcileCas(this::dateSupplier))
                 .compose(state -> state.clusterOperatorSecret(this::dateSupplier))
+                .compose(state -> state.getKafkaClusterDescription())
```

Comment:
Why do we need to move the get on the Kafka cluster description?
Maybe add a comment for the context why this breaks convention.

Reply:
`getKafkaClusterDescription()` defines `kafkaCluster` what is used in `rollingUpdateForNewCaKey()` for `KafkaRoller` ctor.

Reply:
Makes sense, can we add that as a comment to make that intent clear in the reconcile?

Reply:
I wonder how much dangerous this is since creating the description too early can break other things.

Reply:
To be honest, this move kind of smells of inproper separation of responsibilities.
Is there a less heavy handed way of supplying the information needed in `rollingUpdateForNewCaKey`?

Reply:
I had a look at the code and I do not think moving the method would cause any issues (should be confirmed by STs as well).

I agree that this shows deeper issues. I think the problem is that we do Kafka rolling upgrade on multiple places (C renewals, upgrades, regular RUetc.) and not just in one place. That leads to inconsistencies and errors. But I think it is beyond this PR to fix that. We should probably have some issue for it and look at it in the future.